tests/rkt_run_pod_manifest_test.go

```diff
@@ -0,0 +1,381 @@
+			// Override the image's mount point spec.
+			"rkt-test-run-pod-manifest-vol-rw-override.aci",
+			[]string{
+				"--mounts=dir1,path=/dir1,readOnly=true",
+			},
+			&schema.PodManifest{
+				Apps: []schema.RuntimeApp{
+					{
+						Name: baseAppName,
+						App: &types.App{
+							Exec:  []string{"/inspect", "--write-file", "--read-file"},
+							User:  "0",
+							Group: "0",
+							Environment: []types.EnvironmentVariable{
+								{"FILE", "/dir2/file"},
+								{"CONTENT", "host:bar"},
+							},
+							MountPoints: []types.MountPoint{
+								{"dir1", "/dir2", false},
+							},
+						},
+					},
+				},
+				Volumes: []types.Volume{
+					{"dir1", "host", tmpdir, nil},
+				},
+			},
+			true,
+			"<<<host:bar>>>",
+		},
+		{
+			// Simple read after write with volume mounted, no apps in pod manifest.
+			"rkt-test-run-pod-manifest-vol-rw-no-app.aci",
+			[]string{
+				"--exec=/inspect --write-file --read-file --file-name=/dir1/file --content=host:baz",
+				"--mounts=dir1,path=/dir1,readOnly=false",
+			},
+			&schema.PodManifest{
+				Apps: []schema.RuntimeApp{
+					{Name: baseAppName},
+				},
+				Volumes: []types.Volume{
+					{"dir1", "host", tmpdir, nil},
+				},
+			},
+			true,
+			"<<<host:baz>>>",
+		},
+		{
+			// Simple read after write with volume mounted, no apps in pod manifest.
+			// This should succeed even the mount point in image manifest is readOnly,
+			// because it is overrided by the volume's readOnly.
+			"rkt-test-run-pod-manifest-vol-ro-no-app.aci",
+			[]string{
+				"--exec=/inspect --write-file --read-file --file-name=/dir1/file --content=host:zaz",
+				"--mounts=dir1,path=/dir1,readOnly=true",
+			},
+			&schema.PodManifest{
+				Apps: []schema.RuntimeApp{
+					{Name: baseAppName},
+				},
+				Volumes: []types.Volume{
+					{"dir1", "host", tmpdir, &boolFalse},
+				},
+			},
+			true,
+			"<<<host:zaz>>>",
+		},
+		{
+			// Simple read after write with read-only volume mounted, no apps in pod manifest.
+			// This should fail as the volume is read-only.
+			"rkt-test-run-pod-manifest-vol-ro-no-app.aci",
+			[]string{
+				"--exec=/inspect --write-file --read-file --file-name=/dir1/file --content=baz",
+				"--mounts=dir1,path=/dir1,readOnly=false",
+			},
+			&schema.PodManifest{
+				Apps: []schema.RuntimeApp{
+					{Name: baseAppName},
+				},
+				Volumes: []types.Volume{
+					{"dir1", "host", tmpdir, &boolTrue},
+				},
+			},
+			false,
+			`Cannot write to file "/dir1/file": open /dir1/file: read-only file system`,
+		},
+	}
+
+	for i, tt := range tests {
+		patchTestACI(tt.imageName, tt.imagePatches...)
+		hash := importImageAndFetchHash(t, ctx, tt.imageName)
+		defer os.Remove(tt.imageName)
+
+		tt.podManifest.ACKind = schema.PodManifestKind
+		tt.podManifest.ACVersion = schema.AppContainerVersion
+
+		imgName := types.MustACName(tt.imageName)
+		imgID, err := types.NewHash(hash)
+		if err != nil {
+			t.Fatalf("Cannot generate types.Hash from %v: %v", hash, err)
```

Comment:
error/fatal should be if at test fails to pass validation, I prefer to use panic to distinguish set-up or other runtime errors.